### ISSUE TITLE
reload/refresh dataTable not work

### ISSUE THREAD

**Issue by githubhasib** [bllim/laravel4-datatables-package#211]
I wan to reload my data table when my another function call success. i write reload code but bllim not support reload

my code

    <script>

        function statusFunction(id,str) {

            $.ajax({
                url: "user/statusChange",
                type: "post",
                data: { id:id ,currentStatus:str},
                success: function (msg) {
                    //$("#result").html(responseText);

                    if(msg=='success')
                    {
                        $('#articles').DataTable().ajax.reload();

                    }

                }
            });
        }

        $('#articles').dataTable( {
            "bProcessing": true,
            "bServerSide": true,
            "sScrollX": "600",
            "bScrollCollapse": true,
            "sAjaxSource": "user/contentajax",
            "aaSorting": [[ 0, "desc" ]],
            "aoColumns": [
                { 'sWidth': '60px' },
                { 'sWidth': '180px', 'sClass': 'center' },
                { 'sWidth': '180px', 'sClass': 'center' },
                { 'sWidth': '100px', 'sClass': 'center' },
                { 'sWidth': '100px', 'sClass': 'center' },
                { 'sWidth': '100px', 'sClass': 'center' },
                { 'sWidth': '60px', 'sClass': 'center' },
                { 'sWidth': '60px', 'sClass': 'center' }
            ],
            "sPaginationType": "bootstrap"
        });

    </script>

**Comment by yajra** [comment]
I think what you should use is `$('#articles').DataTable().draw(false)` to reload the table without leaving the current page or changing settings status.

**Comment by githubhasib** [comment]
$('#articles').DataTable().draw(false) not working here

**Comment by yajra** [comment]
I am using DT1.10.7 and it's working for me. Paging and search results stays and just reloads the current page. It's like the `fnStandingRedraw` on DT1.9. BTW, what version of DT and Laravel are you using? 

On the other hand, I am using my own fork of this project so I am not really sure on the server side part. But back when I was using this project, that code used to work for me.

**Comment by githubhasib** [comment]
server side is

        $result = DB::table('examples')
            ->select(["examples.id as id", "examples.name as name","examples.price as price","examples.description as description","examples.picture as picture","examples.visibility_status as visibility_status",
                DB::raw("
       case
          when visibility_status = '1'
          then 'Visible'
          when visibility_status = '0'
          then 'Invisible'
          end as visibility_status_text
    ")]);

        return Datatables::of($result)
            ->remove_column('visibility_status_text')
            ->edit_column('visibility_status', ' <span class="label label-info">{{$visibility_status_text}}</span> <button type="button" class="btn btn-warning btn-flat" onclick="statusFunction({{ $id}},{{$visibility_status }})"><i class="fa fa-fw fa-edit"></i></button>')

            ->edit_column('picture', '<a href="#" class="thumbnail"><img src="{{ $picture }}" alt="{{ $name }}" width="65px"></a>')
            ->add_column('edit_delete', '<a href="/test/examples/{{ $id }}" class="btn btn-warning btn-flat"><i class="fa fa-fw fa-edit"></i>Edit</a>
                                        <a href="/test/examples/{{ $id }}" class="btn btn-danger btn-flat"><i class="fa fa-fw fa-times"></i>Delete</a>')

            ->make();


also i am using Laravel 4 and DT version     1.9.4 @yajra

**Comment by yajra** [comment]
I see, since you're using DT1.9.4 then you should use the fnStandingRedraw (http://www.datatables.net/plug-ins/api/fnStandingRedraw) plugin. `DataTable` is only available at v1.10++. Also try inspecting your javascript and check the console for any errors.

**Comment by githubhasib** [comment]
I think this package not support  v1.10++. @MarkVaughn

**Comment by githubhasib** [comment]
working with 

                    if(msg=='success')
                    {

                        var table = $('#articles').dataTable()
                        table.fnStandingRedraw();
                    }

Thanks @yajra

**Comment by yajra** [comment]
Glad to help! BTW, this package supports DT1.10. The `DataTable()` I am referring to was on the javascript side.
